utils/remote_control/navigator_keyboard_control/remote_control_lib/remote_control_lib.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+
+'''
+Remote Control Library: This object contains all of the common variables and
+functions that are shared by the various remote control devices on NaviGator.
+'''
+
+
+import functools
+import itertools
+
+import actionlib
+from geometry_msgs.msg import WrenchStamped
+from navigator_alarm import AlarmBroadcaster
+from navigator_alarm import AlarmListener
+from navigator_msgs.msg import ShooterDoAction, ShooterDoActionGoal
+from navigator_msgs.srv import WrenchSelect
+import rospy
+from std_srvs.srv import Trigger, TriggerRequest
+
+
+__author__ = "Anthony Olive"
+__maintainer__ = "Anthony Olive"
+__email__ = "anthony@iris-systems.net"
+__copyright__ = "Copyright 2016, MIL"
+__license__ = "MIT"
+
+
+class RemoteControl(object):
+
+    def __init__(self, controller_name, wrench_pub=None):
+        self.name = controller_name
+        self.wrench_choices = itertools.cycle(['rc', 'keyboard', 'autonomous'])
+
+        self.alarm_broadcaster = AlarmBroadcaster()
+        self.kill_alarm = self.alarm_broadcaster.add_alarm(
+            name='kill',
+            action_required=True,
+            severity=0
+        )
+        self.station_hold_alarm = self.alarm_broadcaster.add_alarm(
+            name='station_hold',
+            action_required=False,
+            severity=3
+        )
+
+        # rospy.wait_for_service('/change_wrench')
+        self.wrench_changer = rospy.ServiceProxy('/change_wrench', WrenchSelect)
+        self.kill_listener = AlarmListener('kill', self.update_kill_status)
+
+        if (wrench_pub is None):
+            self.wrench_pub = wrench_pub
+        else:
+            self.wrench_pub = rospy.Publisher(wrench_pub, WrenchStamped, queue_size=1)
+
+        self.shooter_load_client = actionlib.SimpleActionClient('/shooter/load', ShooterDoAction)
+        self.shooter_fire_client = actionlib.SimpleActionClient('/shooter/fire', ShooterDoAction)
+        self.shooter_cancel_client = rospy.ServiceProxy('/shooter/cancel', Trigger)
+
+        self.is_killed = False
+        self.is_timed_out = False
+        self.clear_wrench()
+
+    def timeout_check(function):
```

Comment:
Since this is fairly controller specific, it's the controller's job to let the library know that it is timed out. Once this happens, most functions are effectively disabled. If you would like to see an example of this, check out  utils/navigator_gui/navigator_gui/dashboard.py ~line 335.